sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/models/ElevationResult.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.maps.elevation.models;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.models.GeoPosition;
+import com.azure.maps.elevation.implementation.models.Elevation;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.ArrayList;
+import java.util.List;
+
+/** The response from a successful Get Data for Bounding Box API. */
+@Immutable
+public final class ElevationResult {
+    /*
+     * The response for point/points elevation API. The result will be in same
+     * sequence of points listed in request.
+     */
+    @JsonProperty(value = "data", access = JsonProperty.Access.WRITE_ONLY)
+    private List<Elevation> elevations;
+
+    /** Set default ElevationResult constructor to private */
+    private ElevationResult() {}
+
+    /**
+     * Get the elevations property: The response for point/points elevation API. The result will be in same sequence of
+     * points listed in request.
+     *
+     * @return the elevations value
+     */
+    public List<GeoPosition> getElevations() {
+        List<GeoPosition> toreturn = new ArrayList<>();
```

Comment:
would it make sense to cache the result in the constructor (or lazily upon request), so that if customers call `getElevations()` multiple times we don't copy each time?

Reply:
also, would it make sense to return read-only list ?

Reply:
This is rarely used more than once so it would make more sense to let customers know they should re-use the result (if it ever happens) instead of adding a more complicated transform process via the ClassCustomization library for a simple add that may cause other issues

If added this requires adding a private field private List<GeoPosition> positions; that would require a more complex roundabout solution utilizing transformAST since it is modifying auto-generated code:

public List<GeoPosition> getElevations() {
        if (positions == null) {
            positions = new ArrayList<>();
            for (Elevation e : this.elevations) {
                positions.add(
                    new GeoPosition(
                        e.getCoordinate().getLatitude(),
                        e.getCoordinate().getLongitude(),
                        (double) e.getElevationInMeters()));
             }
     }
     return positions;
 }